wagtail/wagtailadmin/views/pages.py

```diff
@@ -17,22 +17,38 @@
 from wagtail.wagtailadmin.forms import CopyForm, SearchForm
 from wagtail.wagtailadmin.utils import send_notification
 from wagtail.wagtailcore import hooks
-from wagtail.wagtailcore.models import Page, PageRevision, get_navigation_menu_items
+from wagtail.wagtailcore.models import (
+    Page, PageRevision, UserPagePermissionsProxy, get_explorable_page_paths, get_navigation_menu_items
+)
 
 
 def explorer_nav(request):
     return render(request, 'wagtailadmin/shared/explorer_nav.html', {
-        'nodes': get_navigation_menu_items(),
+        'nodes': get_navigation_menu_items(request.user),
     })
 
 
 def index(request, parent_page_id=None):
     if parent_page_id:
         parent_page = get_object_or_404(Page, id=parent_page_id)
+
+        if not UserPagePermissionsProxy(request.user).for_page(parent_page).can_explore():
```

Comment:
I think you can use `parent_page.permissions_for_user(request.user)` here instead of `UserPagePermissionsProxy(request.user).for_page(parent_page)`.